_posts/2023-08-02-#28122.md

```diff
@@ -0,0 +1,62 @@
+
+## Questions
+
+1. Did you read the [BIP](https://github.com/bitcoin/bips/pull/1458)? In a few sentences, can you summarize it in your own words?
+
+1. Did you review the [reference implementation](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/reference.py) and [test vectors](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/send_and_receive_test_vectors.json)? Can you think of additional test cases?
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
+
+1. In [5688262](https://github.com/bitcoin/bitcoin/pull/28122/commits/56882622faf469b6f948f79a69c3c8ddbde92ff8), why do we need a custom hash function? Why not just use the default hash function from `libsecp256k1`? (hint: [see this discussion on the BIP](https://github.com/bitcoin/bips/pull/1458#pullrequestreview-1466163601))
+
+1. [ae6019b](https://github.com/bitcoin/bitcoin/pull/28122/commits/ae6019b3a5f99e63b5f23e48fd430147b8500877) and [439e57a](https://github.com/bitcoin/bitcoin/pull/28122/commits/439e57aa88d33e9c61e4c38a37526133a79ab188) add functions for decoding and encoding silent payment addresses, but [src/key_io.cpp](https://github.com/bitcoin/bitcoin/blob/439e57aa88d33e9c61e4c38a37526133a79ab188/src/key_io.cpp) already has a `DestinationEncoder` class and a `DecodeDestination` function. Why can't we just add silent payment addresses to the existing encoder and decoder? More specifically, what is a `CTxDestination`? How is it different from a silent payment address? (hint: `CTxDestination` is defined in [src/script/standards.h](https://github.com/josibake/bitcoin/blob/439e57aa88d33e9c61e4c38a37526133a79ab188/src/script/standard.h))
+
+1. [BIP352](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352.mediawiki#versions) talks about about versioning and forward compatibility. Why is this important? Do you see any issues with the way the PR is written with respect to forwards compatibility?
+
+1. In [`DecodeSilentAddress`](https://github.com/bitcoin/bitcoin/pull/28122/commits/ae6019b3a5f99e63b5f23e48fd430147b8500877#diff-a83ac7cb4a9d60b647db6378ccfad815005724e594bf53f49745b0f40701292fR327) there is a check on the version and data size. In your own words, can you explain what this check is doing? Why is it important?
+
+1. The bulk of the silent payments code is under the wallet directory in [src/wallet/silentpayments.cpp](https://github.com/bitcoin/bitcoin/blob/2dedccebac504b6f3aea68cea6eb7537a16fd8ea/src/wallet/silentpayments.cpp). Is this a good place for the code to live? Why or why not? Can you think of use case where it would be better to have the code outside the wallet?
```

Comment:
Same here, use the mirror repo to avoid the links going dead

Reply:
Fixed